### PR TITLE
Integrated Bootstrap in the User Interface...

### PR DESCRIPTION
**Pull request by MoeweX**
... and made some minor improvements. 

bildschirmfoto 2014-03-06 um 10 51 04
Functionality remains the same
bildschirmfoto 2014-03-06 um 10 51 17

### CODE REVIEW

**Comment by StephanEwen** [comment]
Looks very good, I like it :-)

Before we can merge this, we need to make sure the legal stuff is cleared.

Can you please go though all external libraries / css / javascript / etc that you added and find out the license under which it is distributed? Please post the list here so we get an overview and can double check. It is a bit tedious, but actually really important!

**Comment by MoeweX** [comment]
Most parts are from bootstrap/jquery -> MIT license: https://github.com/twbs/bootstrap/blob/master/LICENSE

*timeline.js:  "Licensed under the Apache License, Version 2.0 (the "License")"
*jcanvas.min.js": MIT license
*jquery.tablesorter.js: "Dual licensed under the MIT and GPL licenses"
*font-awesome: MIT license

I think that should be all

**Comment by StephanEwen** [comment]
What about the images for the icons?

**Comment by MoeweX** [comment]
that is font-awesome: http://fortawesome.github.io/Font-Awesome/

**Comment by StephanEwen** [comment]
Okay, good.

I am a bit confused about the dual license MIT / GPL. What does that mean? Both licenses have very different implications?

**Comment by rmetzger** [comment]
I think it is possible that software has different licenses at the same
time.
For example there are two versions of jackson-core, one with a `-asl` and
one with a `-lgpl` suffix.

On Thu, Mar 6, 2014 at 12:08 PM, Stephan Ewen notifications@github.comwrote:

> Okay, good.
> 
> I am a bit confused about the dual license MIT / GPL. What does that mean?
> Both licenses have very different implications?
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/stratosphere/stratosphere/pull/534#issuecomment-36845341
> .

## 

Robert Metzger, Kontakt: metzgerr@web.de, Mobil: 0171/7424461

File: stratosphere-runtime/resources/web-docs-infoserver/index.html [hunk 1]
```diff
@@ -1,84 +1,168 @@
...
+
+    <!-- Bootstrap core CSS -->
+    <link href="css/bootstrap.css" rel="stylesheet">
+
+    <!-- Add custom CSS here -->
+    <link href="css/sb-admin.css" rel="stylesheet">
+    <link rel="stylesheet" href="font-awesome/css/font-awesome.min.css">
+
+    <!-- Page Specific CSS -->
+    <link rel="stylesheet" href="http://cdn.oesmith.co.uk/morris-0.4.3.min.css">
```
Comment (rmetzger) [inline stratosphere-runtime/resources/web-docs-infoserver/index.html]: is it possible to ship all required CSS and JS files with our code?
This requires the user to have a working internet connection.

File: stratosphere-runtime/resources/web-docs-infoserver/analyze.html [hunk 2]
```diff
@@ -1,88 +1,128 @@
...
+		   </div>
+		</div>
+
+    </div><!-- /#page-wrapper -->
+
+    </div><!-- /#wrapper -->
+
+    <!-- Page Specific Plugins -->
+    <script src="http://cdnjs.cloudflare.com/ajax/libs/raphael/2.1.0/raphael-min.js"></script>
+    <script src="http://cdn.oesmith.co.uk/morris-0.4.3.min.js"></script>
```
Comment (rmetzger) [inline stratosphere-runtime/resources/web-docs-infoserver/analyze.html]: external dependencies.

File: stratosphere-runtime/resources/web-docs-infoserver/js/jobmanagerFrontend.js [hunk 3]
```diff
@@ -35,7 +35,7 @@ $(init());
 function poll(jobId) {
 	$.ajax({ url : "jobsInfo?get=updates&job="+jobId, type : "GET", cache: false, success : function(json) {
 
-		// Call init of no more jobs are running
+		// Call init of no more jobs are running˝
```
Comment (rmetzger) [inline stratosphere-runtime/resources/web-docs-infoserver/js/jobmanagerFrontend.js]: remove the `˝`

**Comment by MoeweX** [comment]
I removed the external dependencies and deleted not used files. If we need them in the future we can add them again (http://startbootstrap.com/sb-admin) @rmetzger

**Comment by rmetzger** [comment]
The interface is beautiful.

But it does not work with YARN (the link in the archive to the `analyze.html` is an absolute path.
(We had this bug already https://github.com/stratosphere/stratosphere/issues/325).

**Comment by fhueske** [comment]
:+1: really like it as well. Nicely done!

**Comment by rmetzger** [comment]
When running a job the sums in the last row do not make sense. They are not the sums of the rows above.

**Comment by StephanEwen** [comment]
I agree with @rmetzger 
If the sums are off, we will get questions about that. Can you have a look at that.

@markus-h Can you also review the pull request?

**Comment by MoeweX** [comment]
Can you please specify in which case the sums are off or include a screenshot? I made some test and I can't see any differences
kmeansiterative

**Comment by MoeweX** [comment]
Changes of my last commit (colored collums):
changes
I put it in it's own commit, because it is not only a bug fix, it is an improvement.

**Comment by rmetzger** [comment]
The YARN bug still exists. (have a look at the URL in the screenshot)

And 16*3 =! 32 (Starting column)

Can you add some space below the "cancel" button ?

wrong

And:

wrong1

**Comment by rmetzger** [comment]
I think it is okay this way.

**Comment by MoeweX** [comment]
It is not, sorry for any inconvenience, I found a bug and I am working on it

**Comment by MoeweX** [comment]
@markus-h  I think I fixed it, the old code is still inside the update function, I am going to delete it when you say my version works fine for you :).

I recognized that the Job-History only contains the last five jobs, is this on purpose?

**Comment by rmetzger** [comment]
Hi,
you can configure the number of jobs stored in the history.

**Comment by MoeweX** [comment]
Ok, good. So everything is fine in that regard.

**Comment by MoeweX** [comment]
I integrated the changes from @qmlmoon in his commits f050b708268b781fd86cadd96aeaf6469ebec7e2 and 7aacbf3994cca859d4097ebea117934899a7b684 because it could not be done automatically.

In addition to that I created a placeholder message in the running jobs table
bildschirmfoto 2014-03-25 um 14 33 50

**Comment by rmetzger** [comment]
Have you fixed the YARN bug?

**Comment by MoeweX** [comment]
Of course!

**Comment by StephanEwen** [comment]
Thanks!

Merged in cc8e16ebd53500c146dd9826436f983d3f243151

**Comment by MoeweX** [comment]
Fine :), but @StephanEwen, why does it say: 
bildschirmfoto 2014-03-26 um 01 07 11

**Comment by StephanEwen** [comment]
I rebased the commit to the head of the master branch. We do that usually to keep a straight and clean commit history. Rebasing the commit changes the hash, because the hash is based on content and preceding commits.

With a changed hash, git does not recognize the branch as merged. But your changes are in, don't worry ;-)

**Comment by StephanEwen** [comment]
BTW: where do you see this message? Are you using some client?

**Comment by MoeweX** [comment]
I see this message on the website ... . Ok, I rebased my changes to the last version I had in the afternoon. So I guess something was changed later on.

**Pull request closed without merging**